packages/odoo/test/Adaptor.test.js

```diff
@@ -178,3 +180,114 @@ describe('delete', () => {
     expect(data).to.eql(true);
   });
 });
+
+describe('searchReadRecord', () => {
+  const makeData = n => {
+    const rows = [];
+    for (let i = 0; i < n; i++) {
+      rows.push({
+        id: i + 1,
+        name: `P${i + 1}`,
+        is_company: true,
+        im_status: 'im_partner',
+      });
+    }
+    rows.push({
+      id: 10001,
+      name: 'NonMatch',
+      is_company: false,
+      im_status: 'im_partner',
+    });
+    rows.push({
+      id: 10002,
+      name: 'VendorOnly',
+      is_company: true,
+      im_status: 'im_vendor',
+    });
+    return rows;
+  };
+
+  it('fetches up to the limit using multiple pages and returns nextOffset=null at the end', async () => {
+    const mockData = makeData(200);
+    const client = new MockOdooWithSearch(mockData);
+
+    setMockClient(client);
+
+    const { data } = await searchReadRecord(
+      'res.partner',
+      { is_company: true },
+      ['name'],
+      {
+        pageSize: 50,
+      }
+    )(state);
+
+    expect(data.totalFetched).to.equal(200);
+    expect(data.rows).to.have.length(200);
+    expect(data.nextOffset).to.equal(null);
+    expect(data.rows[199]).to.deep.equal({ name: 'P200' });
+  });
+
+  it('stops early when data ends before the limit', async () => {
```

Comment:
This isn't quite right because again, the number of items exceeds the limit.

So you have 120 items in the database, but you've passed a `limit` of 200. You're saying "don't give me more than 200 items", and it doesn't, but only by chance.

If you set `limit` lower than the dataset, then this test works